File: PrecomputedPrime.py
import math
n = int(input())
primes=[]
sqr = math.ceil(math.sqrt(n))
arr = list(range(0,sqr+1))
arr[0]=arr[1]=0
i=2
while(i*i<=sqr):
    if(arr[i]!=0):
        for j in range(i*i,sqr+1,i):
            arr[j]=0
    i+=1
arr = list(filter(lambda x: x!=0,arr))
for i in range(len(arr)):
    while(n%arr[i]==0):
        primes.append(arr[i])
        n//=arr[i]
if(n>1):
    primes.append(n)
print(*primes)

File: PrecomputedPrime.cpp
#include<iostream>
#include<cmath>
using namespace std;
int main()
{
    int n;
    cin >> n;
    int sqr = ceil(sqrt(n));
    int primes[sqr],k=0;
    int arr[sqr+1];
    arr[0]=arr[1]=0;
    for(int i=2;i<=sqr;i++)
    {
        arr[i] = 1;
    }
    for(int i=2;i*i<=sqr;i++)
    {
        if(arr[i]==1)
        {
            for(int j=i*i;j<=sqr;j+=i)
            {
                arr[j]=0;
            }
        }
    }
    int y[sqr],j=0;
    for(int i=2;i<=sqr;i++)
    {
        if(arr[i]==1)
        {
            y[j++]=i;
        }
    }

    for(int i=0;i<j;i++)
    {
        while(n%y[i]==0)
        {
            primes[k++] = y[i];
            n/=y[i];
        }
    }
   
    if(n>1)
    {
        primes[k++] = n;
    }
    for(int i=0;i<k;i++)
    {
        cout << primes[i] << " ";
    }
    
}